standalone-metastore/src/main/thrift/hive_metastore.thrift

```diff
@@ -1967,26 +1984,51 @@ service ThriftHiveMetastore extends fb303.FacebookService
       throws(1:AlreadyExistsException o1, 2:NoSuchObjectException o2, 3:InvalidObjectException o3, 4:MetaException o4)
 
   // Schema calls
-  void create_ischema(1:ISchema schema) throws(1:AlreadyExistsException o1,
+  i64 create_ischema(1:ISchema schema) throws(1:AlreadyExistsException o1,
         NoSuchObjectException o2, 3:MetaException o3)
   void alter_ischema(1:string schemaName, 2:ISchema newSchema)
         throws(1:NoSuchObjectException o1, 2:MetaException o2)
-  ISchema get_ischema(1:string schemaName) throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  ISchema get_ischema_by_name(1:string schemaName) throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  ISchema get_ischema(1:i64 schemaId) throws (1:NoSuchObjectException o1, 2:MetaException o2)
   void drop_ischema(1:string schemaName)
         throws(1:NoSuchObjectException o1, 2:InvalidOperationException o2, 3:MetaException o3)
 
-  void add_schema_version(1:SchemaVersion schemaVersion)
+  i64 add_schema_version(1:ISchemaVersion schemaVersion)
         throws(1:AlreadyExistsException o1, 2:NoSuchObjectException o2, 3:MetaException o3)
-  SchemaVersion get_schema_version(1: string schemaName, 2: i32 version)
+  ISchemaVersion get_schema_version(1: string schemaName, 2: i32 version)
+        throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  ISchemaVersion get_schema_latest_version(1: string schemaName)
         throws (1:NoSuchObjectException o1, 2:MetaException o2)
-  SchemaVersion get_schema_latest_version(1: string schemaName)
+  ISchemaVersion get_schema_version_by_id(1:i64 schemaVersionid)
         throws (1:NoSuchObjectException o1, 2:MetaException o2)
-  list<SchemaVersion> get_schema_all_versions(1: string schemaName)
+
+  list<ISchemaVersion> get_schema_versions_by_name_and_fingerprint(1:string schemaName, 2:string fingerPrint)
+        throws (1:NoSuchObjectException o1, 2:MetaException o2)
+  list<ISchemaVersion> get_schema_all_versions(1: string schemaName)
         throws (1:NoSuchObjectException o1, 2:MetaException o2)
   void drop_schema_version(1: string schemaName, 2: i32 version)
         throws(1:NoSuchObjectException o1, 2:MetaException o2)
   FindSchemasByColsResp get_schemas_by_cols(1: FindSchemasByColsRqst rqst)
         throws(1:MetaException o1)
+
+  void add_schema_branch(1:ISchemaBranch schemaBranch)
```

Comment:
Some of the feedback on the earlier patch that I posted to add the schema registry calls is that rather than using arguments in each of the thrift calls we should instead build a request and response structure for each call.  This makes it easier to make changes to the calls in the future.  You'll need to do the same for the new schema branch calls.  You can take a look at the calls to add and get schemas and schema versions for examples.